server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -513,8 +537,94 @@ public ClusterMetadataManifest getLatestClusterMetadataManifest(String clusterNa
         return fetchRemoteClusterMetadataManifest(clusterName, clusterUUID, latestManifestFileName);
     }
 
+    /**
+     * Fetch the previous cluster UUIDs from remote state store and return the most recent valid cluster UUID
+     *
+     * @param clusterName The cluster name for which previous cluster UUID is to be fetched
+     * @return Last valid cluster UUID
+     */
+    public String getLatestClusterUUID(String clusterName) {
+        try {
+            Set<String> clusterUUIDs = getAllClusterUUIDs(clusterName);
+            Map<String, ClusterMetadataManifest> latestManifests = getLatestManifestForAllClusterUUIDs(clusterName, clusterUUIDs);
+            List<String> validChain = createClusterChain(latestManifests);
+            if (validChain.isEmpty()) {
+                return ClusterState.UNKNOWN_UUID;
+            }
+            return validChain.get(0);
+        } catch (IOException e) {
+            logger.error("Exception while fetching previous UUIDs for cluster name: {}", clusterName);
```

Comment:
only throw exception, that would logged where it will get caught right, dont need another log. Do cross check also.

Reply:
applies at other places as well.